src/ape/api/query.py

```diff
@@ -0,0 +1,79 @@
+from typing import Any, Dict, List, Literal, Optional, Union
+
+import pandas as pd
+from pydantic import BaseModel
+
+from ape.types import AddressType
+from ape.utils import abstractdataclass, abstractmethod
+
+Query = Union["BlockQuery", "AccountQuery", "ContractEventQuery", "ContractMethodQuery"]
+
+
+class BaseQuery(BaseModel):
+    type: str  # Used as discriminator
+    columns: List[str]
+    filter_args: Dict[str, Any]
```

Comment:
We will probably need to have a bigger discussion around how to handle this filter_args.

Reply:
Yeah... lol

Tbh, the `start_block...end_block` and `start_nonce...end_nonce` params are fulfilling this role, but other things might be available to a dataframe (TBD research item post this PR)

Can probably remove for now!